### PR TITLE
Resolve #43 -- Code rework

### LINKED ISSUE #43: Rework code and design change
Some parts of the code need to be reworked. Also, the design has to be changed to one similar to our test enviroment client

### PR DESCRIPTION
**Pull request by Deudly**
Code rework, new design and summoner icons

Refs #43

### CODE REVIEW

File: src/ui/components/lobby/lobbyComponent.ts [hunk 1]
```diff
@@ -28,77 +28,97 @@ Vue.partial("lobby-player-list", require("./partials/playerList.html"));
...
+
+        NetworkServiceStatic.currentLobbyConnection.emit("host");
+    }
+    data() {
+        return {
+            message: "",
+            messages: [],
+            lobby: NetworkServiceStatic.currentLobby,
+            network: NetworkService,
+            isHost: this.isHost,
```
Comment (molenzwiebel) [inline src/ui/components/lobby/lobbyComponent.ts:79]: This won't do anything.

File: src/ui/components/lobby/lobbyComponent.ts [hunk 2]
```diff
@@ -28,77 +28,97 @@ Vue.partial("lobby-player-list", require("./partials/playerList.html"));
...
+
+    leave() {
+        NetworkServiceStatic.leaveLobby();
+        this.$router.go("/lobbies");
     }
-    
-    get playerSettings() {
-        return this.settings.playerSettings;
+
+    iconByUser(user: { idServer: number}) {
```
Comment (molenzwiebel) [inline src/ui/components/lobby/lobbyComponent.ts]: `iconForUser` would be a better name, fix the spacing, get a complete type in here.

File: src/ui/components/lobbyList/lobbyListComponent.ts [hunk 3]
```diff
@@ -6,34 +6,57 @@ const Vue = <vuejs.VueStatic>require("vue");
...
     created() {
         // Mainly for debugging. Redirects if theres no connection.
         if (!NetworkService.currentConnection) {
-            this.$router.go("/loading");
+            this.$router.go("/login");
         }
     }
     
     data() {
+        this.canCreateLobby = true;
```
Comment (molenzwiebel) [inline src/ui/components/lobbyList/lobbyListComponent.ts]: This should go in the data initializer.

File: src/ui/components/lobbyList/lobbyListComponent.ts [hunk 4]
```diff
@@ -6,34 +6,57 @@ const Vue = <vuejs.VueStatic>require("vue");
...
         return {
-            username: "Player",
-            network: NetworkService
+            network: NetworkService,
+            staticService: StaticService,
+            lobbySelected: undefined,
```
Comment (molenzwiebel) [inline src/ui/components/lobbyList/lobbyListComponent.ts]: This should be null, not undefined.

File: src/ui/components/lobbyList/lobbyListComponent.ts [hunk 5]
```diff
@@ -6,34 +6,57 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+            canCreateLobby: this.canCreateLobby
         };
     }
     
+    select(lobby: lobby.LobbyListItem) {
+        this.lobbySelected = lobby;
```
Comment (molenzwiebel) [inline src/ui/components/lobbyList/lobbyListComponent.ts]: `selectedLobby` would be a better name.

File: src/ui/components/lobbySettings/lobbySettingsComponent.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            selectedIcon: ModalComponent.params[0],
+            settings: SettingServiceStatic,
+        };
+    }
+
+    get hostSettings() {
+        return this.settings.hostSettings;
+    }
+    
+    finish(arg) {
```
Comment (molenzwiebel) [inline src/ui/components/lobbySettings/lobbySettingsComponent.ts]: `arg` has no type.

File: src/ui/components/login/loginComponent.ts [hunk 7]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
-    
+
     private username: string;
-    private password: string;
-    
+    private host: string;
+    private path: string;
+    private port: string;
+    private state: string;
+    private icon: any;
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: This should be a proper type.

File: src/ui/components/login/loginComponent.ts [hunk 8]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+
     data() {
+        if (localStorage.getItem("path") != undefined && localStorage.getItem("path") != "") {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `this.path = localStorage.getItem("path") || "Path to LoL 4.20";`

File: src/ui/components/login/loginComponent.ts [hunk 9]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+            this.path = localStorage.getItem("path");
+        } else {
+            this.path = "Path to LoL 4.20";
+        }
+        if (localStorage.getItem("host") != undefined && localStorage.getItem("host") != "") {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `if (localStorage.getItem("host"))`

File: src/ui/components/login/loginComponent.ts [hunk 10]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+            this.host = localStorage.getItem("host");
+        }
+        if (localStorage.getItem("port") != undefined && localStorage.getItem("port") != "") {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `if (localStorage.getItem("port"))`

File: src/ui/components/login/loginComponent.ts [hunk 11]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+            this.port = localStorage.getItem("port");
+        }
+        if (localStorage.getItem("name") != undefined && localStorage.getItem("name") != "") {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `if (localStorage.getItem("name"))`

File: src/ui/components/login/loginComponent.ts [hunk 12]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+            this.username = localStorage.getItem("name");
+        }
+        var iconId = 0;
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `var`

File: src/ui/components/login/loginComponent.ts [hunk 13]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+        if (localStorage.getItem("icon") != undefined && localStorage.getItem("icon") != "") {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts]: `if (localStorage.getItem("icon"))`

File: src/ui/components/login/loginComponent.ts [hunk 14]
```diff
@@ -6,32 +6,84 @@ const Vue = <vuejs.VueStatic>require("vue");
...
+        });
+    }
+
+    openIconPicker() {
+        ModalComponent.present("select-icon", this.icon).then(c => {
+            this.icon = c;
+        });
+    }
+
+    indexByiconId(id: number): number {
```
Comment (molenzwiebel) [inline src/ui/components/login/loginComponent.ts:83]: This should have a better name. I have no clue what it does.

File: src/ui/css/main.less [hunk 15]
```diff
@@ -93,6 +93,161 @@ body {
...
+    color: #f0e6d2;
+    letter-spacing: 0;
+    text-transform: uppercase;
+    background: -webkit-repeating-linear-gradient(#e5cda0 0,#e5cda0 10px,#79622b 60px,#79622b 75px);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+}
+
+.nav-bar {
+    background: #2A628F url(https://lolstatic-a.akamaihd.net/frontpage/apps/prod/lcu_alpha_website/en_US/c0dcb26e1ba53437859331627d5e2f01dfda818e/assets/img/bgs/magic-page-top.jpg);
```
Comment (molenzwiebel) [inline src/ui/css/main.less]: This should ideally be something locally hosted.

File: src/ui/css/main.less [hunk 16]
```diff
@@ -93,6 +93,161 @@ body {
...
+    border-image: linear-gradient(to bottom,#c0ab71,#645127) 1 stretch;
+    border-width: 3px;
+    text-align: center;
+    font-family: "Beaufort for LOL";
+    font-size: 15px;
+    padding: 0 5px;
+    white-space: nowrap;
+}
+
+@login-background-image: url("https://lolstatic-a.akamaihd.net/frontpage/apps/prod/lcu_alpha_website/en_US/c0dcb26e1ba53437859331627d5e2f01dfda818e/assets/img/bgs/overview-section.jpg");
```
Comment (molenzwiebel) [inline src/ui/css/main.less]: Again, host locally.

File: src/ui/services/networkService.ts [hunk 17]
```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
     /** Current Socket.IO connection to central server. */
     currentConnection: SocketIOClient.Socket;
 
+    /** List of all users. */
+    users: user.User[];
+
+    me: user.User;
+
+    imHost: boolean;
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `isHost`

File: src/ui/services/networkService.ts [hunk 18]
```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
...
     }
 
-    connectToCentral(url: string): Promise<void> {
+    connectToCentral(url: string, username = "Unknown", iconId: number): Promise<void> {
         return new Promise<void>((resolve, reject) => {
             this.currentConnection = io.connect(url, { reconnection: false });
             this.currentConnection.on("connect", resolve);
-
-            const [add, update, remove] = this.buildListUpdater("lobbylist", this.lobbies, {
+            var userInfo = {
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `var`

File: src/ui/services/networkService.ts [hunk 19]
```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
...
-                name: "name",
-                creator: "creator",
-                playerLimit: "10",
-                gamemodeName: "gameMode",
-                requirePassword: "false",
-                address: "address",
-                port: "port"
-            }
-            this.currentConnection.emit('lobby.create', options);
+            this.currentConnection.on("lobby.list", function(lobbiesList){
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: Arrow function, `for (const lobby of lobbiesList)`

File: src/ui/services/networkService.ts [hunk 20]
```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
...
+                }
+            });
             this.currentConnection.emit('lobby.list');
-            this.currentConnection.on("lobby.list", function(lobbies){
-                var count = 0;
-                while (count <= lobbies.length){
-                    add(lobbies[count]);
-                    count++;
+            
+            this.currentConnection.on('user.list', function(clientsList){
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: Arrow function, `for (const lobby of clientsList)`

File: src/ui/services/networkService.ts [hunk 21]
```diff
@@ -76,24 +98,54 @@ export class NetworkService extends EventEmitter {
...
         }
 
-        this.currentUsername = username;
         return new Promise((resolve, reject) => {
-            this.currentLobbyConnection = io.connect(item.address + ":" + item.port, {reconnection: false, 'forceNew': true});
+            this.currentLobbyConnection = io.connect(item.address + ":" + item.port, {reconnection: true, forceNew: true});
+            this.currentLobbyConnection.on("lobby-connect", (c) => this.handleLobbyConnect(c, resolve, reject));
             this.currentLobbyConnection.on("connect", () => {
-                this.currentLobbyConnection.emit("lobby-connect", { name: username, password: password });
+                this.currentLobbyConnection.emit("lobby-connect", { idServer: this.me.id, username: this.me.username, password: "" });
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `idServer` -> `serverId`

File: src/ui/services/networkService.ts [hunk 22]
```diff
@@ -76,24 +98,54 @@ export class NetworkService extends EventEmitter {
...
     public leaveLobby() {
+        console.log("Lobby left");
         this.currentLobbyConnection.disconnect();
+        this.currentLobbyConnection.close();
+        this.currentLobby = undefined;
+        this.currentLobbyConnection = undefined;
+    }
+
+    public createLobby(): Promise<any> {
+        var options = {
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `var`

File: src/ui/services/networkService.ts [hunk 23]
```diff
@@ -76,24 +98,54 @@ export class NetworkService extends EventEmitter {
...
+            name: this.me.username + "'s lobby",
+            playerLimit: "10",
+            gamemodeName: "Capture the Deudly",
+            password: ""
+        }
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `;`

File: src/ui/services/networkService.ts (hunk 17 above)
Comment (molenzwiebel) [reply src/ui/services/networkService.ts]: Also, this doesn't seem to be used anywhere.

File: src/ui/services/networkService.ts [hunk 24]
```diff
@@ -158,26 +220,23 @@ export class NetworkService extends EventEmitter {
...
-                spellOne: ["spell1id", id => StaticService.summonerSpells.filter(x => x.id === id)[0]],
-                spellTwo: ["spell2id", id => StaticService.summonerSpells.filter(x => x.id === id)[0]]
+                isHost: "isHost"
             });
-            function startGame(gameServerPort){
+            function startGame(data){
                 //Start the game with the port
                 console.log("Starting LoL...")
-                var configContent = readConfigData();
                 var args = [
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `var`

File: src/ui/services/networkService.ts [hunk 25]
```diff
@@ -158,26 +220,23 @@ export class NetworkService extends EventEmitter {
...
+                username: "username",
                 team: ["teamId", id => this.currentLobby.teams.filter(x => x.id === id)[0]],
-                champion: ["championId", id => StaticService.champions.filter(x => x.id === id)[0]],
-                skinIndex: "skinIndex",
-                spellOne: ["spell1id", id => StaticService.summonerSpells.filter(x => x.id === id)[0]],
-                spellTwo: ["spell2id", id => StaticService.summonerSpells.filter(x => x.id === id)[0]]
+                isHost: "isHost"
             });
-            function startGame(gameServerPort){
+            function startGame(data){
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: Nested functions are never a good idea.

File: src/ui/services/networkService.ts [hunk 26]
```diff
@@ -158,26 +220,23 @@ export class NetworkService extends EventEmitter {
...
                     "8394",
                     "LoLLauncher.exe",
                     "",
-                    "127.0.0.1 " + gameServerPort + " 17BLOhi6KZsTtldTsizvHg== " + playerId
+                    "127.0.0.1 " + data.gameServerPort + " 17BLOhi6KZsTtldTsizvHg== " + data.playerId
                 ];
-                execFile.execFile(configContent.pathToLolExe, 
-                args, {cwd: configContent.pathToLolFolder, maxBuffer: 1024 * 90000},
+                execFile(localStorage.getItem("path") + "/League of Legends.exe", 
+                args, {cwd: localStorage.getItem("path"), maxBuffer: 1024 * 90000},
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: Either leave this on a single line or indent it properly, this is one statement that looks like two.

File: src/ui/services/networkService.ts [hunk 27]
```diff
@@ -212,6 +271,13 @@ export class NetworkService extends EventEmitter {
                     this.emit("chat", d, m.sender, m.message);
                 });
             });
+            this.currentLobbyConnection.on("host", data => {
+                if(data.isHost) {
+                    this.emit("host", true); 
+                } else {
+                    this.emit("host", false); 
+                }
```
Comment (molenzwiebel) [inline src/ui/services/networkService.ts]: `this.emit("host", data.isHost)`

File: src/vendor/protocol.d.ts [hunk 28]
```diff
@@ -153,7 +153,7 @@ declare namespace lobby {
         /** Name of the lobby. */
         name: string;
         /** Creator of the lobby. */
```
Comment (molenzwiebel) [inline src/vendor/protocol.d.ts]: Update the documentation.

File: src/vendor/protocol.d.ts [hunk 29]
```diff
@@ -164,4 +164,24 @@ declare namespace lobby {
...
+        username: string;
+        /** ID of the icon */
+        iconId: number;
+    }
+
+    interface Icon {
+        id: number;
+
+        iconURL: string;
+    }
```
Comment (molenzwiebel) [inline src/vendor/protocol.d.ts]: Documentation.

**Review by molenzwiebel — CHANGES_REQUESTED** [review]
I didn't review your styles since you are wildly inconsistent with how you format your less files. Mixed indentation, spaces before commas and no spaces before commas, lines over 300 characters long just to mention a few things. I recommend you get a proper editor to work on this project, since you're obviously not using one (or you're not using it properly).

File: src/ui/components/lobby/lobbyComponent.ts (hunk 2 above)
Comment (Deudly) [reply src/ui/components/lobby/lobbyComponent.ts]: d

File: src/ui/components/lobbyList/lobbyListComponent.ts (hunk 4 above)
Comment (Deudly) [reply src/ui/components/lobbyList/lobbyListComponent.ts]: d

File: src/ui/components/lobbyList/lobbyListComponent.ts (hunk 5 above)
Comment (Deudly) [reply src/ui/components/lobbyList/lobbyListComponent.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 9 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 8 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 10 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 12 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 13 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/services/networkService.ts (hunk 17 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 18 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 19 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 20 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 21 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 22 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 23 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 25 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 24 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 26 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/ui/services/networkService.ts (hunk 27 above)
Comment (Deudly) [reply src/ui/services/networkService.ts]: d

File: src/vendor/protocol.d.ts (hunk 29 above)
Comment (Deudly) [reply src/vendor/protocol.d.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 11 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/vendor/protocol.d.ts (hunk 28 above)
Comment (Deudly) [reply src/vendor/protocol.d.ts]: d

File: src/ui/components/login/loginComponent.ts (hunk 7 above)
Comment (Deudly) [reply src/ui/components/login/loginComponent.ts]: d

File: src/ui/components/lobbySettings/lobbySettingsComponent.ts (hunk 6 above)
Comment (Deudly) [reply src/ui/components/lobbySettings/lobbySettingsComponent.ts]: d

File: src/ui/components/lobbyList/lobbyListComponent.ts (hunk 3 above)
Comment (Deudly) [reply src/ui/components/lobbyList/lobbyListComponent.ts]: d

**Pull request closed without merging**